### PR TITLE
feat: misc updates

### LINKED ISSUE #76: Styling / Display Updates
- better colors (#33)
- smaller max-width on posts (CenteredPage.svelte)
- `lang` field for markdown
- manage images in markdown
- remark-autolink-headings and remark-toc
- (maybe) improve codeblocks
- (maybe) cleanup global styles

### PR DESCRIPTION
**Pull request by Eggrror404**
closes #76 

basically random updates / fixes

- made max-width smaller in `CenteredPage.svelte`
- added lang field for markdown (and a notice for posts not in the current language)
- added custom `Image` components for image display in markdown
- used `rehype-slug` for heading ids
- added numbers to codeblocks

now we still have to figure out the colors

### CODE REVIEW

**Comment by Eggrror404** [comment]
@HeyLoon review plz

**Comment by Eggrror404** [comment]
btw you can use normal images
and they will (probably) scale properly now

**Comment by HeyLoon** [comment]
@Eggrror404 I made some change, leave a message if everything is OK.

**Comment by HeyLoon** [comment]
> btw you can use normal images and they will (probably) scale properly now

I will use it next time :)

File: src/messages/ja.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
-  "alert_caution": "Caution",
-  "alert_important": "Important",
-  "alert_note": "Note",
-  "alert_tip": "Tip",
-  "alert_warning": "Warning",
+  "alert_caution": "注意",
```
Comment (HeyLoon) [inline src/messages/ja.json:2]: Let me explain here wahtever chines or japanese translation.
I searched Cambridge Dictionary to check the difference between caution, important, note, tip and warning.
My conclusion is Important>Warning>Caution>Tip>Note, so juist 逕依各國國情之慣用語酌予排序。

Comment (Eggrror404) [reply src/messages/ja.json:2]: To me its
Caution > Warning > Important > Tip > Note

And thats how github do it i think
https://github.com/orgs/community/discussions/16925

**Review by HeyLoon — APPROVED** [review]
:D

**Comment by Eggrror404** [comment]
we should clean up the translation messages
some of them isn't even used lol

**Comment by Eggrror404** [comment]
thats probably not a good idea
image

**Comment by Eggrror404** [comment]
> thats probably not a good idea image

are we fixing or not?

**Comment by HeyLoon** [comment]
> > thats probably not a good idea image
> 
> are we fixing or not?

I think it's better than the previous one

**Comment by Eggrror404** [comment]
> > > thats probably not a good idea image
> > 
> > 
> > are we fixing or not?
> 
> I think it's better than the previous one

wouldn't it break mobile view

**Comment by HeyLoon** [comment]
> > > > thats probably not a good idea image
> > > 
> > > 
> > > are we fixing or not?
> > 
> > 
> > I think it's better than the previous one
> 
> wouldn't it break mobile view

image
not really bad

**Pull request merged**